src/components/time-picker/time-picker.tsx

```diff
@@ -593,6 +596,7 @@ export class TimePicker {
         localizedSecondSuffix,
         localizedMeridiem
       } = localizeTimeStringToParts(value, this.locale);
+      this.formatParts = getformatToParts(value, this.locale);
       this.localizedHour = localizedHour;
```

Comment:
Can you compute the meridiem order here instead of on every render? Storing the parts may no longer be needed after this.